### ISSUE TITLE
Charset isn't set in ContentType

### ISSUE THREAD

**Issue by LukaJCB** [hseeberger/akka-http-json#142]
I'm using version `1.15.0` of akka-http-json and `0.7.1` of circe.

By default the ContentType of a response is set to `application/json`, meaning that the client doesn't know which charset to use and has trouble displaying e.g. umlauts. I'd like to be able to set the charset or have a sensible default like utf-8. Right now when I write a very simple application that does nothing but respond with a JSON:

```

case class Foo(s: String)

import io.circe.generic.auto._
import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._

val route =
  path("hello") {
     get {
       complete(Foo("München"))
      }
   }
```

And I send a Request to such a Server with the following Headers:

```
GET /hello HTTP/1.1
Host: localhost:8080
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
DNT: 1
Accept-Charset: utf-8
Accept-Encoding: gzip, deflate, sdch, br
Accept-Language: en-US,en;q=0.8,de;q=0.6
```

I receive the following response:

```
HTTP/1.1 200 OK
Content-Length: 16
Content-Type: application/json
Date: Wed, 12 Apr 2017 14:02:10 GMT
Server: akka-http/10.0.5

{
    "s": "MÃ¼nchen"
}
```

I would expect the Response header to include `charset=utf-8`.

We could very easily just set the Response to utf-8 by default. I could even create a PR if desired.

**Comment by hseeberger** [comment]
Very interesting! Let me play with this a little, which might take some time, because I'm on vacation.

**Comment by LukaJCB** [comment]
I've found that you can very simply solve this problem by specifying that the `contentType` of the `HttpEntity` be `` `application/json`.withParams(Map("charset" -> "utf-8"))`` instead of just `` `application/json` ``.

**Comment by hseeberger** [comment]
Where? Client or server?

**Comment by LukaJCB** [comment]
On the server. Something like 

```
val body = Foo("München").asJson.noSpaces
val entity = HttpEntity(`application/json`.withParams(Map("charset" -> "utf-8")), body)
complete(entity)
```

**Comment by hseeberger** [comment]
After reading up on JSON (e.g. [RFC 7159](https://tools.ietf.org/html/rfc7159#page-9) and [SO](http://stackoverflow.com/questions/9254891/what-does-content-type-application-json-charset-utf-8-really-mean#9254967)) I think that the current implementation is just fine, because `application/json` is defined as `MediaType.WithFixedCharset` (of course UTF-8 is used) and UTF-8 is defined as the default encoding for JSON, hence setting the charset in the response header explicitly is redundant. The client needs to be changed to assume UTF-8 if not indicated otherwise.

**Comment by LukaJCB** [comment]
I see your points, however, I'm not sure if that's the way it should be dealt with. I don't think we can change browser internals to assume UTF-8 (maybe there's something I don't know about, the only thing I can think about is add a request header to accept `utf-8` only). 
Moreover, if I look at any other public api out there they all include the response header `Content-Type: application/json; charset=utf-8`, I concur that "all the others do it" is not the greatest argument, but I believe it shouldn't stray to far from the others for the sake of compatibility.

**Comment by hseeberger** [comment]
You're speaking about a browser, but browsers are for HTML. Have you tried your example with a HTTP client, e.g. curl or httpie?

The closest others in this case are Akka and the other libraries supporting JSON encoding. AFAIK none sends that header.

**Comment by hseeberger** [comment]
I tried to reproduce this issue but could not. Neither with httpie nor with Safari.

Now here's a rude assumption: when you implemented the server, are you sure the source files are UTF-8 encoded? Please double check.

**Comment by LukaJCB** [comment]
Sorry for getting back to you late, I was on vacation as well. 
Now here's something odd, I double checked the encoding of the source file and it is indeed UTF-8, however I checked again with httpie and this time it seems to have worked, even though I could have sworn it did not the last time I tried. Browser(Chrome) still gives me wrong answer though. I concur with your point that they are for HTML. I was assigned to fix the issue by a colleague of mine, so I'm going to have to check with him for which client he was using. Thanks again for your time! :)
